homeassistant/components/snoo/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Snoo Sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from python_snoo.containers import SnooData, SnooStates
+
+from homeassistant.components.sensor import (
+    EntityCategory,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    StateType,
+)
+from homeassistant.const import UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SnooConfigEntry
+from .coordinator import SnooCoordinator
+from .entity import SnooDescriptionEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class SnooSensorEntityDescription(SensorEntityDescription):
+    """Describes a Snoo binary sensor."""
+
+    value_fn: Callable[[SnooData], bool]
+
+
+SENSOR_DESCRIPTIONS: list[SnooSensorEntityDescription] = [
+    SnooSensorEntityDescription(
+        key="state",
+        translation_key="state",
+        value_fn=lambda data: data.state_machine.state.name,
+        device_class=SensorDeviceClass.ENUM,
+        options=[e.name for e in SnooStates],
+    ),
+    SnooSensorEntityDescription(
+        key="time_left",
+        translation_key="time_left",
+        value_fn=lambda data: data.state_machine.time_left,
+        device_class=SensorDeviceClass.DURATION,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+    ),
```

Comment:
How much does this update? Does it just count down? Would it make sense to just make this a timestamp?

Reply:
Perhaps a timestamp makes sense?

But the weird thing about it is that it is going to pause for a bit before it ever goes down - so I'm not sure if a timestamp would work as if HA started counting down, then the api reset it back to it's original it might be weird.

Reply:
Right, but that means it will be set to 480. And leave the impression that it will be another 8 minutes. But after 2 minutes, that's not correct anymore. But with a timestamp sensor, it's still correct, because that one will show that it finishes in 6 minutes

Reply:
So you're saying turn the int into a timestamp and use that?

Reply:
I think that works better yes, because that one stays actual. Otherwise it will be a "2 minutes ago it was another 8 minutes". And if you make the sensor display the end time of the current program, you will always see "it will end in 6 minutes"

Reply:
And it will go -1 when there is no active timer, should I just set it to None in that case?

Reply:
Sounds about right!

Reply:
That works way better - you were definitely right!